test.disabled/ejabberd_tests/tests/rest_client_SUITE.erl

```diff
@@ -521,3 +555,198 @@ assert_json_room_sse_message(Expected, Received) ->
+            escalus:send(Alice, escalus_stanza:roster_add_contact(Bob,
+                         [], <<"Bob">>)),
+            PushReqB = escalus:wait_for_stanza(Alice),
+            escalus:assert(is_roster_set, PushReqB),
+            escalus:send(Alice, escalus_stanza:iq_result(PushReqB)),
+            escalus:assert(is_iq_result, escalus:wait_for_stanza(Alice)),
+            %% Alice sends subscribed presence
+            escalus:send(Alice,
+                         escalus_stanza:presence_direct(
+                             escalus_client:short_jid(Bob),
+                             <<"subscribed">>)),
+            % now check Bob's roster
+            timer:sleep(100),
+            {?OK, R4} = gett("/contacts", BCred),
+            Result4 = decode_maplist(R4),
+            [Res4] = Result4,
+            #{jid := AliceJID, subscription := <<"to">>,
+                ask := <<"none">>} = Res4,
+            ok
+        end
+    ),
+    ok.
+
+add_contact_and_be_invited(Config) ->
+    escalus:fresh_story(
+        Config, [{alice, 1}, {bob, 1}],
+        fun(Alice, Bob) ->
+            AliceJID = escalus_utils:jid_to_lower(
+                escalus_client:short_jid(Alice)),
+            BCred = credentials({bob, Bob}),
+            % bob has empty roster
+            {?OK, R} = gett("/contacts", BCred),
+            Res = decode_maplist(R),
+            [] = Res,
+            % adds Alice
+            AddContact = #{jid => AliceJID},
+            {?NOCONTENT, _} = post(<<"/contacts">>, AddContact,
+                                   BCred),
+            % and she is in his roster, with empty status
+            {?OK, R2} = gett("/contacts", BCred),
+            Result = decode_maplist(R2),
+            [Res2] = Result,
+            #{jid := AliceJID, subscription := <<"none">>,
+              ask := <<"none">>} = Res2,
+            % and he received a roster push
+            Push = escalus:wait_for_stanza(Bob),
+            escalus:assert(is_roster_set, Push),
+            %% she adds him and invites
+            escalus:send(Alice, escalus_stanza:roster_add_contact(Bob,
+                         [],
+                         <<"Bobek">>)),
+            escalus:assert_many([is_roster_set, is_iq_result],
+                                escalus:wait_for_stanzas(Alice, 2)),
+            escalus:send(Alice,
+                         escalus_stanza:presence_direct(
+                             escalus_client:short_jid(Bob),
+                             <<"subscribe">>)),
+            escalus:assert(is_roster_set, escalus:wait_for_stanza(Alice)),
+            escalus:assert(is_presence_with_type, [<<"subscribe">>],
+                           escalus:wait_for_stanza(Bob, 1)),
+            % now check Bob's roster, and it is empty...
+            {?OK, R4} = gett("/contacts", BCred),
+            Result4 = decode_maplist(R4),
+            [] = Result4,
+            % because it is stated in RFC3921, 8.2.6, and implemented
```

Comment:
How about make the roster get function configurable? In case of XMPP user it works as is, in case of REST call it will return all items if with state `{none, in}`?